qbraid_qir/squin/visitor.py

```diff
@@ -0,0 +1,440 @@
+                    qbit_getitem = frame.push(
+                        py.GetItem(self.qreg, index_ssa)  # pylint: disable=too-many-function-args
+                    )
+                    self.qubit_ssa_map[qid] = qbit_getitem.result
+            else:
+                # Create individual qubits using qubit.new() and store SSA values directly in map
+                for qid in range(self.num_qubits):
+                    squin_qubit = frame.push(
+                        func.Invoke(  # pylint: disable=unexpected-keyword-arg, too-many-function-args
+                            (), callee=qubit.new
+                        )
+                    )
+                    self.qubit_ssa_map[qid] = squin_qubit.result
+
+            self.visit(state)
+
+            if compactify:
+                Walk(CFGCompactify()).rewrite(frame.curr_region)
+
+            region = frame.curr_region
+
+        return region
+
+    def visit(self, state: lowering.State[pyqir.Module]) -> lowering.Result:
+        """Visit a PyQIR module.
+
+        Args:
+            state (lowering.State[pyqir.Module]): The state of the visitor.
+
+        Returns:
+            lowering.Result: The result of the visitor.
+        """
+        # There could be multiple basic blocks in the entry point
+        if len(self.entry_point.basic_blocks) < 1:  # type: ignore
+            raise InvalidSquinInput("No basic blocks found in entry point, expected at least 1")
+        for block in self.entry_point.basic_blocks:  # type: ignore
+            self.visit_statement(state, block)
+
+    def visit_statement(
+        self, state: lowering.State[pyqir.Module], statement: Any
+    ) -> lowering.Result | None:
+        """Visit a PyQIR statement.
+
+        Args:
+            state (lowering.State[pyqir.Module]): The state of the visitor.
+            statement (Any): The statement to visit.
+
+        Returns:
+            lowering.Result: The result of the visitor.
+        """
+        visit_map: dict[type, Callable[[lowering.State[pyqir.Module], Any], lowering.Result]] = {
+            pyqir.Call: self.visit_call,
+            pyqir.BasicBlock: self.visit_basic_block,
+            pyqir.Constant: self.visit_constant,
+            pyqir.FloatConstant: self.visit_floatConstant,
+            pyqir.IntConstant: self.visit_intConstant,
+        }
+        visitor_function = visit_map.get(type(statement))
+        if visitor_function:
+            return visitor_function(state, statement)
+
+        return None
+
+    def visit_basic_block(
+        self, state: lowering.State[pyqir.Module], block: pyqir.BasicBlock
+    ) -> lowering.Result:
+        """Visit a PyQIR basic block.
+
+        Args:
+            state (lowering.State[pyqir.Module]): The state of the visitor.
+            block (pyqir.BasicBlock): The basic block to visit.
+
+        Returns:
+            lowering.Result: The result of the visitor.
+        """
+        if not isinstance(block, pyqir.BasicBlock):
```

Comment:
This instance check is not required as block is guaranteed to be `pyqir.BasicBlock` given how we are [calling this method](https://github.com/qBraid/qbraid-qir/pull/252/files#diff-a4d52835de56c1b122439b79b0130cb1e911c252fef93deee967ff4e2091deccR318-R320). If this was user-facing, then this check would make more sense. 

Similar comment for [`visit_call`](https://github.com/qBraid/qbraid-qir/pull/252/files#diff-a4d52835de56c1b122439b79b0130cb1e911c252fef93deee967ff4e2091deccR353) as well